providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagResolution.java

```diff
@@ -0,0 +1,229 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import com.google.protobuf.NullValue;
+import dev.openfeature.contrib.providers.flagd.strategy.ResolveStrategy;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.MutableStructure;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Value;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static dev.openfeature.contrib.providers.flagd.Config.CACHED_REASON;
+import static dev.openfeature.contrib.providers.flagd.Config.CONTEXT_FIELD;
+import static dev.openfeature.contrib.providers.flagd.Config.FLAG_KEY_FIELD;
+import static dev.openfeature.contrib.providers.flagd.Config.REASON_FIELD;
+import static dev.openfeature.contrib.providers.flagd.Config.STATIC_REASON;
+import static dev.openfeature.contrib.providers.flagd.Config.VALUE_FIELD;
+import static dev.openfeature.contrib.providers.flagd.Config.VARIANT_FIELD;
+
+final class FlagResolution {
```

Comment:
A bit of a nit, but since this has protobuf stuff in it, I think it should potentially be under the `grpc/` dir.